plugins/org.eclipse.xtext/src/org/eclipse/xtext/serializer/analysis/SemanticSequencerNfaProvider.java

```diff
@@ -63,6 +63,20 @@ public SemNfa(ISemState starts, ISemState stops) {
 		}
 
 		@Override
+		public int hashCode() {
+			if (hashCode == -1)
+				hashCode = new NfaUtil().hashCode(this, GET_ASSIGNED_GRAMMAR_ELEMENT);
```

Comment:
It appears that the hashCode and equalsIgnoreCase are externalized but only used from here and used with the same keyfunction. Is that a temporary situation or would it make sense to inline the logic here and thereby reduce the complexity?